src/types/PLC/xml-data/data-types/struct/data-type-struct.ts

```
@@ -0,0 +1,20 @@
+import { baseTypes } from '@root/shared/data'
+import { z } from 'zod'
+
+const dataTypeStructSchema = z.object({
+  variable: z.array(
+    z.object({
+      '@name': z.string(),
+      type: z.object(baseTypes.reduce((acc, type) => ({ ...acc, [type]: z.string().optional() }), {})).optional(),
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)